### ISSUE TITLE
private should mean private

### ISSUE THREAD

**Issue by marler8997** [marler8997/dc#41]
In my opinion, `private` should mean private to the class/struct, not to the module.  I think `module` ownership is a fine idea, but should be have it's own visibility.

```D
struct Foo
{
    private int x;
    package int y;
}

Foo foo;
foo.x; // error, x is not visible
foo.y; // ok
```

**Comment by wilzbach** [comment]
I never considered this a big issue. On the contrary, I quite like that for testing you can introspect into the class if you want to.
(though I don't have a strong feeling about this)

**Comment by marler8997** [comment]
Yeah, white box testing is good and this problem of accessing private fields is a common one in any language. We could have the language grant visibility to all private fields if it's inside a `unittest` block or `version(unittest)`.

**Comment by aliak00** [comment]
Or have a `fileprivate` scope or `public/protected/private/package/module`. Or `package(this)` to mean module. Or `private(this)` to mean "real" private and don't change the current private.
